src/Shared/UnitTests/MockEngine.cs

```diff
@@ -399,8 +413,7 @@ internal void AssertLogContains(string contains)
         {
             if (_upperLog == null)
             {
-                _upperLog = _log;
-                _upperLog = _upperLog.ToUpperInvariant();
+                _upperLog = _log.ToString().ToUpperInvariant();
```

Comment:
I'm going to leave this since I'm not making it worse, but yeah, that might bite us in the future :(